packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -498,7 +498,8 @@ class _NavigationRailState extends State<NavigationRail> with TickerProviderStat
                           labelTextStyle: widget.selectedIndex == i ? selectedLabelTextStyle : unselectedLabelTextStyle,
                           padding: widget.destinations[i].padding,
                           onTap: () {
-                            widget.onDestinationSelected!(i);
+                            if(widget.onDestinationSelected != null)
+                              widget.onDestinationSelected!(i);
```

Comment:
What about this then? :)

```suggestion
                            if (widget.onDestinationSelected != null)
                              widget.onDestinationSelected!(i);
```

Would be interesting to get insights on the style choice from a team member. Especially the null-assertion (bang) after the null check seems off.